### ISSUE TITLE
Parser UTC offset logic is inverted

### ISSUE THREAD

**Issue by atttx123** [dateutil/dateutil#70]
my python-dateutil version: ```2.4.1```

i am in the time zone: ```CST(UTC+8)```

when i use dateutil, it return different with other tools

pyhton-dateutil:
```
In [21]: parse("2015-03-31 00:01 UTC").astimezone(cst)
Out[21]: datetime.datetime(2015, 3, 31, 8, 1, tzinfo=tzlocal())

In [22]: parse("2015-03-31 00:01 UTC+1").astimezone(cst)
Out[22]: datetime.datetime(2015, 3, 31, 9, 1, tzinfo=tzlocal())

In [23]: parse("2015-03-31 00:01 UTC-1").astimezone(cst)
Out[23]: datetime.datetime(2015, 3, 31, 7, 1, tzinfo=tzlocal())
```

linux date command:
```
yu at debian in ~ 
(! 4076)-> date -d "2015-03-31 00:01 UTC"
Tue Mar 31 08:01:00 CST 2015
yu at debian in ~ 
(! 4076)-> date -d "2015-03-31 00:01 UTC+1"
Tue Mar 31 07:01:00 CST 2015
yu at debian in ~ 
(! 4077)-> date -d "2015-03-31 00:01 UTC-1"
Tue Mar 31 09:01:00 CST 2015
```

**Comment by atttx123** [comment]
i think it is the ```parse``` return a wrong tzoffset

```
In [21]: parse("2015-03-31 00:01 UTC-1")
Out[21]: datetime.datetime(2015, 3, 31, 0, 1, tzinfo=tzoffset(None, 3600))

In [22]: parse("2015-03-31 00:01 UTC+8")
Out[22]: datetime.datetime(2015, 3, 31, 0, 1, tzinfo=tzoffset(None, -28800))
```

**Comment by pganssle** [comment]
@atttx123 Given that tzlocal() is `CST`, are you sure that that is the wrong answer? I looked into something similar for issue #64, and came to the conclusion that `tzlocal()` and a local time zone are equivalent.

**Comment by pganssle** [comment]
@atttx123 I'm going to assume that you are satisfied and close this as not a bug. If this is mistaken let me know and I can reopen.

**Comment by atttx123** [comment]
@pganssle the point is not the timezone, its the ```parse``` function itself

when the it is ```00:00 UTC+1```, my local time should be ```07:00 UTC+8```
but dateutil return:

```
In [6]: parse("2015-03-31 00:01 UTC+1").astimezone(cst)
Out[6]: datetime.datetime(2015, 3, 31, 9, 1, tzinfo=tzfile('/usr/share/zoneinfo/Asia/Shanghai'))
``` 

it return ```09:00``

**Comment by atttx123** [comment]
on linux ,you can use this cmd to check time:
```
date -d "2015-03-31 00:01 UTC+1"
```

i am sure dateutil return a wrong answer

**Comment by pganssle** [comment]
Ah, sorry, I missed that. My bad. I'll look into it.

**Comment by pganssle** [comment]
Hm... This will require careful thought on how to address this, unfortunately. It seems that this was a deliberate choice, as seen in [this comment](https://github.com/dateutil/dateutil/blob/master/dateutil/parser.py#L803). Interestingly, another comment in the `tz` module specifically [mentions that `tzstr` has the opposite behavior](https://github.com/dateutil/dateutil/blob/master/dateutil/tz.py#L599).

I don't understand why this behavior has not been noticed before.

**Comment by atttx123** [comment]
i think the ```parse``` should behave same with other tools, like ```date``` command, otherwise it will make   people confuse

**Comment by pganssle** [comment]
@atttx123 I agree in general, but it's important to determine if this is a bug that no one has noticed or if this is the well-known behavior of the library. If we want to switch the logic to the more common version, we may want an intermediate version that raises a deprecation warning.

**Comment by atttx123** [comment]
@pganssle if it becomes a feature, this one should highlight in document, comment in code is not enough

**Comment by pganssle** [comment]
Thinking about this a bit more, it seems like dateutil is doing the right thing. Consider that a timestamp "March 3, 2014 00:00:00 UTC+1" is equivalent to "March 2, 2014 23:00:00 UTC"

**Comment by atttx123** [comment]
```+1``` is the calculate or it is the timezone,
for me , ```UTF+8``` is the timezone, i never think it is the ```UTF``` plus ```8``` before

**Comment by pganssle** [comment]
Sorry, that was a half-formed thought, I meant to click "cancel" but I accidentally posted it instead.

**Comment by atttx123** [comment]
maybe this one is more persuasiveness: ```UTC+01:00```

**Comment by atttx123** [comment]
@pganssle if in dateutil +1 means plus one hour, why +01:30 not plus the 30 minute

**Comment by pganssle** [comment]
@atttx123 OK, I've done a bit of googling, and it turns out that this is actually [a feature of POSIX time-zones](http://www.pixelbeat.org/docs/linux_timezones/) that the trailing time has an inverted sense, which has historically led to a lot of confusion.

I will think about how to move forward. If there are datetimestamps out there using UTC offsets in the non-POSIX sense (I assume there are), and there's no way to disambiguate them from context, then maybe a `POSIX` flag in the `parserinfo` is the right way to go.

**Comment by spumer** [comment]
Same issue:
```
>>> # stanard library
>>> dt = datetime.datetime.strptime('2015-06-15T11:19:57 +0500', '%Y-%m-%dT%H:%M:%S %z')
>>> str(datetime.timezone(dt.utcoffset()))
'UTC+05:00'
>>>
>>> # dateutil library
>>> dt2 = dateutil.parser.parse('2015-06-15T11:19:57 UTC+05:00')
>>> str(datetime.timezone(dt2.utcoffset()))
'UTC-05:00'
```

**Comment by zed** [comment]
Perhaps, a boolean parameter `posix_style` that will default to `False` eventually could be introduced. Related: [Timezone offset sign reversed by Python `dateutil`?](http://stackoverflow.com/q/31078749/4279).

**Comment by yohplala** [comment]
Hello,
Is there any update on this issue?
I am sorry, I encountered this trouble as well. I added an example in
https://github.com/pandas-dev/pandas/issues/30518

When resolving the offset with a timezone that has DST in a panda dataframee, and using these timestamps as indexes, this reversed behavior causes trouble has it creates duplicate indexes.
Please, is there any workaround?
I thank you in advance for your help.
Bests,
Pierre

**Comment by pganssle** [comment]
@yohplala No, there is no update at the moment. The parser is a bit of a hard nut to crack. I haven't thought about it in a while but it's really down to coming up with a better way to configure the parser than we have now without adding a bunch of confusing and potentially conflicting flags.

**Comment by zenoprod** [comment]
i encountered this problem too.
when i coding 
`pd.to_datetime('Jul 1, 2021 12:00:01 AM UTC-7')`
it get 

>  Timestamp('2021-07-01 00:00:01+0700', tz='pytz.FixedOffset(420)')

i delete the **UTC-7** and use .tz_localize('America/Los_Angeles') to solve this problem.

**Comment by personalcomputer** [comment]
With regard to workarounds, let me share the preprocessing solution that I am using successfully as a generic workaround:

```python
preprocessed_datetime_str = re.sub(r'(?:GMT|UTC)([+\-]\d+)', r'\1', datetime_str)
dt = dateutil.parser.parse(preprocessed_datetime_str)
```

This converts all offsets like "UTC+0600" or "GMT+0100" into just "+0600" or "+0100", respectively, which causes dateutil to then parse the offset as expected.

**Comment by dansebcar** [comment]
The [current documentation](https://dateutil.readthedocs.io/en/stable/parser.html) on the subject is misleading. The [first reference](https://www.cl.cam.ac.uk/~mgk25/iso-time.html#zone) says UTC offsets east of the zero meridian are positive, and west of the zero meridian are negative.

Is a PR adding a note to the documentation about this exception welcome?
